lib/ash/changeset/changeset.ex

```diff
@@ -1436,6 +1436,7 @@ defmodule Ash.Changeset do
   def prepare_changeset_for_action(changeset, action, opts) do
     changeset
     |> Map.put(:action, action)
+    |> Map.put(:action_type, action.type || :update)
```

Comment:
```suggestion
    |> Map.put(:action_type, action.type)
```

It shouldn't ever be possible to have an action without a type